scim2.yaml

```diff
@@ -1675,104 +1753,573 @@ paths:
           schema:
-            $ref: '#/definitions/ErrorSchemaNotFound'
-
-  /ServiceProviderConfig:
+            $ref: '#/definitions/RoleSearchErrorResponseObject'
+  /v3/Roles/{id}:
     get:
       tags:
-        - ServiceProviderConfig Endpoint
-      summary: Get Service Provider Config
+        - Roles Endpoint
+      summary: Get Role by ID
       description: |
-        This API returns the service provider's configuration details.
+        This API returns the role details of a particular role using its unique ID.
 
-        <b>Permission required:</b>
+        <b>Scope(Permission) required:</b>
 
-            * No permissions required
-      operationId: getServiceProviderConfig
+            *internal_role_mgt_view
+      operationId: getRolebyIdV3
       produces:
         - application/scim+json
-      parameters: []
+      parameters:
+        - name: id
+          in: path
+          description: Unique ID of the resource type.
+          required: true
+          type: string
+        - name: attributes
+          in: query
+          description: SCIM defined attributes parameter.
+          required: false
+          type: string
+        - name: excludedAttributes
+          in: query
+          description: SCIM defined excludedAttribute parameter.
+          required: false
+          type: string
       responses:
         200:
-          description: Schema is found
+          description: Valid role is found
           schema:
-            $ref: '#/definitions/SPConfigResponse'
+            $ref: '#/definitions/RoleGetResponseObjectV3'
         401:
           description: Unauthorized
           schema:
             $ref: '#/definitions/ErrorUnauthorized'
+        403:
+          description: Forbidden
+          schema:
+            $ref: '#/definitions/ErrorForbidden'
         404:
-          description: Schema not found
+          description: Valid role is not found
           schema:
-            $ref: '#/definitions/ErrorSchemaNotFound'
+            $ref: '#/definitions/ErrorNoRoleAvailable'
+        500:
+          description: Internal Server Error
+          schema:
+            $ref: '#/definitions/ErrorInternalServerError'
+    put:
+      tags:
+        - Roles Endpoint
+      summary: Update Role - PUT
+      description: |
+        This API updates the role's display name and permissions, and returns the updated role details using a PUT operation.
 
+        <b>Scope(Permission) required:</b>
 
+            * internal_role_mgt_delete
+      operationId: updateRoleV3
+      consumes:
+        - application/scim+json
+      produces:
+        - application/scim+json
+      parameters:
+        - name: id
+          in: path
+          description: Unique ID of the resource type.
+          required: true
+          type: string
+        - in: body
+          name: body
+          required: false
+          schema:
+            $ref: '#/definitions/RolePutRequestObjectV3'
```

Comment:
This Patch operation should support role name / permission updation.
but the sample has group/user update example.
<img width="878" alt="Screenshot 2025-06-20 at 20 40 36" src="https://github.com/user-attachments/assets/434e556d-559b-44e5-ae30-0dcbf8c5614a" />

Reply:
Fixed in the latest swagger file: https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/645/files#diff-98e4e20cf8479992cac1588ac0b3fffd88f268237d207b39e4d594c2796a5aa6L1311